Makefile: Show better error message when podman/docker command fails

Hi,

there could be a more meaningful error message if the podman/docker command (CONTAINER_CMD) fails.

Example:
If 'podman version' does not work, because the machine is not up and running, 'make build' fails with:

build  --tag samba-container:latest --tag quay.io/samba.org/samba-server:latest -f images/server/Dockerfile.fedora images/server
make: build: No such file or directory
make: *** [.build.server] Error 1

It would be good to see the podman error to know what actually does not work properly and be able to address this.

Thanks!

Cheers,
Karolin

--- a/Makefile
+++ b/Makefile
@@ -5,6 +5,16 @@ endif
 ifeq ($(CONTAINER_CMD),)
 	CONTAINER_CMD:=$(shell docker version >/dev/null 2>&1 && echo docker)
 endif
+# handle the case where podman is present but is (defaulting) to remote and is
+# not not functioning correctly. Example: mac platform but no 'podman machine'
+# vms are ready
+ifeq ($(CONTAINER_CMD),)
+	CONTAINER_CMD:=$(shell podman --version >/dev/null 2>&1 && echo podman)
+ifneq ($(CONTAINER_CMD),)
+$(warning podman detected but 'podman version' failed. \
+	this may mean your podman is set up for remote use, but is not working)
+endif
+endif
 
 BUILD_CMD:=$(CONTAINER_CMD) build $(BUILD_OPTS)
 PUSH_CMD:=$(CONTAINER_CMD) push $(PUSH_OPTS)

